unit_tests/edge_kernels/UnitTestMomentumABLWallFuncEdge.C

```diff
@@ -20,9 +20,10 @@ TEST_F(MomentumABLKernelHex8Mesh, NGP_abl_wall_func)
 {
   if (bulk_.parallel_size() > 1) return;
 
+  const std::string meshSpec;
   const bool doPerturb = false;
   const bool generateSidesets = true;
-  fill_mesh_and_init_fields(doPerturb, generateSidesets);
+  fill_mesh_and_init_fields(meshSpec, doPerturb, generateSidesets);
```

Comment:
I find it really unintuitive and weird to declare `meshSpec` as an empty string and then pass that to a method. I'd to go look at the method to understand what is going on, and I see why you did it. Why not have an overload for `fill_mesh_and_init_fields` 

```C++
// Old behavior
void fill_mesh_and_init_fields(bool doPeturb, bool generateSidesets); 
{
  const std::string meshSpec = "generated:1x1x" + std::to_string(bulk.parallel_size());
  fill_mesh_and_init_fields(meshSpec, doPerturb, generateSidesets);
}

void fill_mesh_and_init_fields(std::string meshSpec, bool doPerturb, bool generateSidesets);
{
 // implementation
}
```

Now only the second method needs to be overridden by the subclasses. 

void

Reply:
We could do that.   But then there are two functions to have virtual overrides all the way down the subclasses.

Reply:
@overfelt can you explain a bit more? I was thinking that since the first method was implemented in terms of the secoyit wouldn't need to be overridden as often by sub classes. Am I overlooking something?